### ISSUE TITLE
Language toggle accessibility updates

### ISSUE THREAD

**Issue by rsmithadhoc** [department-of-veterans-affairs/vets-design-system-documentation#3564]
## Description

Results from a11y review of the language toggle.

## Details

- Some related issues that may be able to be solved with some updated HTML:
    - Context for users of assistive tech:
        - It's hard to immediately tell that these links are related and what they do.
        - Assistive tech users won't know what language is currently selected.
    - Proposed HTML
        ```html
            <div role="group" aria-label="Language selection">
                <button aria-pressed="true">English</button>
                <button aria-pressed="false">Español</button>
                <button aria-pressed="false">Tagalog</button>
            <div>
        ```
- Since this is a user control and not an inline link, I believe we should make the touch target at least 24x24, but we'd prefer even larger at 48x48. Currently, the links are only 20px in height.
- Only having bold text as the visual indicator may not be enough for visually impaired users.
    - This may require some input from design. I initially reviewed the design but didn't catch this. **Edit:** See the results of the design ticket https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3605 
    - Some ideas:
        - A background on the selected language.
          ![Image](https://github.com/user-attachments/assets/86073c52-3271-40ef-aae5-12846c92f024)
        - An icon on the selected language.
          ![Image](https://github.com/user-attachments/assets/d6b314bb-a20c-434f-a371-0f53bcc1f188)

## Tasks
- [ ] Implement accessibility changes.
- [ ] Make updates based on the revised design
- [ ] Add a link to Storybook (https://design.va.gov/storybook/?path=/docs/components-va-language-toggle--docs) on the guidance page (https://design.va.gov/components/language-toggle) 

## Acceptance Criteria
- [ ] Accessibility issues addressed.
- [ ] Design has been updated and reviewed by designer
- [ ] Guidance page has a link to Storybook

## Estimating
Provide your estimate of 1, 2, 3, 5, 8 or 13 
@Andrew565 - 3
@ataker - 
@harshil1793 -  3
@it-harrison - 3
@jamigibbs - 3
@micahchiang -
@powellkerry - 3
@rmessina1010 - 3
@rsmithadhoc -

**Comment by ataker** [comment]
[Link to Figma](https://www.figma.com/design/afurtw4iqQe6y4gXfNfkkk/VADS-Component-Library?node-id=10077-912&p=f)

**Comment by jamigibbs** [comment]
Reopening for some additional work that needs to be done for adjusting the font-size.

- https://github.com/department-of-veterans-affairs/component-library/pull/1502

**Comment by jeana-adhoc** [comment]
Some of Ryan's concerns have been addressed, or attempted to be addressed, some remain.

Here are my recommendations for next steps:

- Aria-label that wraps the group, should add teh selected language.  ie: `aria-label="Language selected: English"` this helps provide additional context.

- currently `aria-current` is set to `language` which, is [not a valid `aria-current` prop](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-current).   This should be `aria-current="true"`

- There was a design ticket #3605 to update the design, but it didn't add in a secondary affordance to identify the current selection like Ryan suggested. Looking at the previous ticket it seems like there was further discussion had and Ryan is on board with the current design. @babsdenney @danbrady and I have talked about doing a clean up audit round of segmented button, tabs, side nav to align all of the various states. I think the work to update the design can be done at that point.

**Comment by jeana-adhoc** [comment]
Tickets created!
